### ISSUE TITLE
Check that all expired tokens return a 401 instead of a 403

### ISSUE THREAD

**Issue by mvdan** [TykTechnologies/tyk#782]
In https://github.com/TykTechnologies/tyk/pull/781, we fixed the following line in `middleware_key_expired_check.go`:

    return errors.New("Key has expired, please renew"), 403

That line should be a `401`, which it now is.

Once the fix is merged, we should double-check that all forms of expired tokens correctly return 401 instead of 403. This includes JWTs, for example.

**Comment by ryanwalls** [comment]
@mvdan Any updates on this?  We are using some libraries that are verifying authentication (auth0) and it only works with 401s.

**Comment by mvdan** [comment]
Well, it is not a priority until we encounter any that can be proven wrong. I opened this issue precisely because doing an exhaustive check would take some time, as there is quite a bit of code and a few libraries involved.

If you do find any status codes that are wrong, or would like to contribute in any way, please feel free to just say so. Otherwise, there isn't any update at the moment.

**Comment by ryanwalls** [comment]
@mvdan Thanks.  Yeah, we'll just hack the client side to work with 403s.

**Comment by mvdan** [comment]
If you do find any status codes that are wrong, and can describe the scenario here, we can fix them :)

**Comment by ryanwalls** [comment]
@mvdan The main one is that expired tokens should ideally return 401s instead of 403s.  Currently when I send an expired token it returns a 403 from Tyk.

**Comment by mvdan** [comment]
Well, if you mean the one coming from the line I gave above, it is already fixed in master (which is the upcoming 2.4). Otherwise, I'm not sure what expired token error you mean in particular.

**Comment by ryanwalls** [comment]
@mvdan How do I know when 2.4 is released for Tyk Cloud customers?

**Comment by ConsM** [comment]
@ryanwalls you can subscribe on the [Releases section](https://community.tyk.io/c/releases) and get notified for new posts.
